netqasm/sdk/memmgr.py

```diff
@@ -0,0 +1,129 @@
+from __future__ import annotations
+
+from itertools import count
+from typing import Dict, List, Set
+
+from netqasm.lang import operand
+from netqasm.lang.encoding import REG_INDEX_BITS, RegisterName
+from netqasm.lang.parsing.text import parse_register
+from netqasm.sdk.futures import Array
+from netqasm.sdk.qubit import Qubit
+
+
+class MemoryManager:
+    def __init__(self) -> None:
+        # All qubits active for this connection
+        self._active_qubits: List[Qubit] = []
+
+        # Registers for looping etc.
+        # These are registers that are for example currently hold data and should
+        # not be used for something else.
+        # For example a register used for looping.
+        self._active_registers: Set[operand.Register] = set()
+
+        self._used_meas_registers: Dict[operand.Register, bool] = {
+            operand.Register(RegisterName.M, i): False for i in range(16)
+        }
+
+        # Registers to return
```

Comment:
This comment is not adding something to the variable name. Maybe some explaining text what this registers holds or otherwise delete the comment